### DISCUSSION TITLE
Rearrange comments in a PR review so that they are in the order of the file/source code

### DISCUSSION THREAD

**Discussion by Popov72** [community/community#44905] [Pull Requests]
### Select Topic Area

Product Feedback

### Body

Hey!

It would be nice to have an extra button when submitting a PR review that would reorder the comments so that they appear in the "Conversation" view in the same order as in the "Changed Files" view.

Quite often I make comments on file A, then make further comments on files B/C/D/..., and finally add additional comments on file A. Currently, these last comments will appear after all my comments on files B/C/D/... and not just after the comment on file A.

The order in which the comments appear in the "Changed Files" view is the order I would like users to see, but unfortunately I can't force them to use the "Changed Files" view when they read my review...

**Comment by andreahlert** [comment]
Hey 🙂
Completely understandable. When feedback order doesn’t follow the file flow, reviews can feel fragmented. Aligning comments with source order would reduce cognitive load.

One thing to consider: should the ordering follow file structure strictly, or also respect chronological discussion threads? Different reviewers might value different mental models.

**Reply by Efrem-Braun** [reply]
I would think it should follow file structure strictly. All the comments a reviewer posts are posted as a whole, so I don't know why there would be a need to care about earlier chronological discussions.
